### ISSUE TITLE
Dialyzer does not pass

### ISSUE THREAD

**Issue by joseph-lozano** [commanded/commanded#313]
Currently, dialyzer does not pass in commanded. 

When using commanded (v1.0.0-rc1) in a project, the following dialyzer issues appear.

```
lib/simple_pricing/match_application.ex:2:callback_type_mismatch
Type mismatch for @callback child_spec/1 in Commanded.EventStore.Adapter behaviour.

Expected type:
[
  {_, {atom(), atom(), :undefined | [any()]}, :permanent | :temporary | :transient,
   :brutal_kill | timeout(), :supervisor | :worker, :dynamic | [atom()]}
  | %{
      :id => _,
      :start => {atom(), atom(), :undefined | [any()]},
      :modules => :dynamic | [atom()],
      :restart => :permanent | :temporary | :transient,
      :shutdown => :brutal_kill | timeout(),
      :type => :supervisor | :worker
    }
]

Actual type:
[atom(), ...]

________________________________________________________________________________
lib/simple_pricing/match_application.ex:2:callback_type_mismatch
Type mismatch for @callback child_spec/1 in Commanded.PubSub.Adapter behaviour.

Expected type:
[
  {_, {atom(), atom(), :undefined | [any()]}, :permanent | :temporary | :transient,
   :brutal_kill | timeout(), :supervisor | :worker, :dynamic | [atom()]}
  | %{
      :id => _,
      :start => {atom(), atom(), :undefined | [any()]},
      :modules => :dynamic | [atom()],
      :restart => :permanent | :temporary | :transient,
      :shutdown => :brutal_kill | timeout(),
      :type => :supervisor | :worker
    }
]

Actual type:
[{Registry, [{_, _}, ...]}, ...]

________________________________________________________________________________
lib/simple_pricing/match_application.ex:2:callback_type_mismatch
Type mismatch for @callback child_spec/1 in Commanded.Registration.Adapter behaviour.

Expected type:
[
  {_, {atom(), atom(), :undefined | [any()]}, :permanent | :temporary | :transient,
   :brutal_kill | timeout(), :supervisor | :worker, :dynamic | [atom()]}
  | %{
      :id => _,
      :start => {atom(), atom(), :undefined | [any()]},
      :modules => :dynamic | [atom()],
      :restart => :permanent | :temporary | :transient,
      :shutdown => :brutal_kill | timeout(),
      :type => :supervisor | :worker
    }
]

Actual type:
[{Registry, [{_, _}, ...]}, ...]
```

**Comment by slashdotdash** [comment]
@joseph-lozano I have a PR to fix the Dialyzer warnings (#317). I'll include this change in v1.0.0 release.

**Comment by slashdotdash** [comment]
Fixed by #317.
